lightning/src/ln/chan_utils.rs

```diff
@@ -1607,6 +1613,62 @@ impl CommitmentTransaction {
 		&self.htlcs
 	}
 
+	/// Returns the index of the revokeable output in the built transaction, if any exists.
+	/// There are two cases where this may return `None`:
+	/// - The balance of the revokeable output is below the dust limit (only found on commitments
+	/// early in the channel's lifetime, i.e. before the channel reserve is met).
+	/// - This commitment was created before LDK 0.0.117. In this case, the
+	/// commitment transaction previously didn't contain enough information to locate the
+	/// revokeable output.
+	pub fn revokeable_output_index(&self) -> Option<usize> {
+		let revokeable_redeemscript = get_revokeable_redeemscript(
+			&self.keys.revocation_key,
+			self.to_broadcaster_delay?,
+			&self.keys.broadcaster_delayed_payment_key,
+		);
+		let revokeable_p2wsh = revokeable_redeemscript.to_v0_p2wsh();
+		let outputs = &self.built.transaction.output;
+		outputs.iter().enumerate()
+			.find(|(_, out)| out.script_pubkey == revokeable_p2wsh)
+			.map(|(idx, _)| idx)
+	}
+
+	/// Helper method to build an unsigned justice transaction spending from an output
+	/// on this transaction to a destination script. Use [`Self::revokeable_output_index`] to get
+	/// the index of the output to spend to create a valid justice transaction. Fee estimation
+	/// assumes the transaction is spending a revoked output, accounting for the expected
+	/// witness data that will be added when signed.
+	///
+	/// This method will error if the given fee rate results in a fee greater than the value
+	/// of the output being spent.
+	pub fn build_justice_tx(&self, output_idx: u32, feerate_per_kw: u64, lock_time: u32, destination_script: Script)
```

Comment:
Went with `build_to_local_justice_tx` to stay consistent with how it's referred to in the spec. Also left `revokeable_output_index` public in case a user wants to build the tx themselves